crypto/s2n_ecdsa.c

```diff
@@ -32,8 +35,29 @@
 #include "crypto/s2n_openssl.h"
 #include "crypto/s2n_pkey.h"
 
+
 #define S2N_ECDSA_TYPE 0
 
+S2N_RESULT s2n_unsafe_ecdsa_get_mut(const struct s2n_ecdsa_key *ecdsa_key, EC_KEY **out_ec_key) {
```

Comment:
we now store EC_KEY as const and its expected to be immutable. however some OSSL functions need a mutable key. in those cases we can cast to a non-const but its an unsafe operation and we better have a `Safety` comment stating why its ok.

```
struct s2n_ecdsa_key {
    const EC_KEY *ec_key;
}
```

Reply:
The "get_mut" would be totally intuitive if this was Rust, but I worry about the readability of using it in the C. How about "s2n_unsafe_ecdsa_get_copy", "s2n_unsafe_ecdsa_get_mutable_copy", or "s2n_unsafe_ecdsa_get_non_const_copy"?

Reply:
`copy` is not true in ossl < 3 since its only in ossl3 that we get back a pre-cached copy. I think the more important bit is that we assume the key is supposed to be `const/immutable` and if you want to get a `not const/mutable` key then you better check all the assumptions or not do it.